### DISCUSSION TITLE
How I'm implementing Scheduled searches on API????

### DISCUSSION THREAD

**Discussion by NSH531** [CrowdStrike/falconpy#698] [Q&A] [answered]
TNX

**Comment by crowdstrikedcs** [comment, marked as answer]
Hi @NSH531 

The API documentation for Scheduled Reports can be found [here](https://falconpy.io/Service-Collections/Scheduled-Reports.html)

These APIs will allow you to query for, run, and download report data. Below is a sample flow

1. Create a scheduled search in the [Falcon UI](https://falcon.crowdstrike.com/documentation/213/scheduled-searches#managing-scheduled-searches)
2. Use FalconPy to [query ](https://falconpy.io/Service-Collections/Scheduled-Reports.html#scheduled_reports_query )and [run](https://falconpy.io/Service-Collections/Scheduled-Reports.html#scheduled_reports_launch) the scheduled report by its id. You can also wait until the scheduled report runs normally. 
3. [Gather](https://falconpy.io/Service-Collections/Scheduled-Reports.html#scheduled_reports_get) the report details.

Code examples for this are included in the above link. I would suggest viewing the documentation in the Falcon UI as well for specifics on creating scheduled searches as this cannot be performed over API at the moment. 

Let us know if you have any follow up on this.

**Comment by NSH531** [comment]
I want  to create a scheduled search from API
Can you access it via FALCONPY's api?

--
nate
+972559870641

‫בתאריך יום ג׳, 21 ביוני 2022 ב-18:15 מאת ‪Dixon Styres‬‏ <‪
***@***.***‬‏>:‬

> Hi @NSH531 <https://github.com/NSH531>
>
> The API documentation for Scheduled Reports can be found here
> <https://falconpy.io/Service-Collections/Scheduled-Reports.html>
>
> These APIs will allow you to query for, run, and download report data.
> Below is a sample flow
>
>    1. Create a scheduled search in the Falcon UI
>    <https://falcon.crowdstrike.com/documentation/213/scheduled-searches#managing-scheduled-searches>
>    2. Use FalconPy to query
>    <https://falconpy.io/Service-Collections/Scheduled-Reports.html#scheduled_reports_query>and
>    run
>    <https://falconpy.io/Service-Collections/Scheduled-Reports.html#scheduled_reports_launch>
>    the scheduled report by its id. You can also wait until the scheduled
>    report runs normally.
>    3. Gather
>    <https://falconpy.io/Service-Collections/Scheduled-Reports.html#scheduled_reports_get>the
>    report details.
>
> Code examples for this are included in the above link. I would suggest
> viewing the documentation in the Falcon UI as well for specifics on
> creating scheduled searches as this cannot be performed over API at the
> moment.
>
> Let us know if you have any follow up on this.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/CrowdStrike/falconpy/discussions/698#discussioncomment-2995025>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABNBJWJ4CCQ3LEYJUK4RW3LVQHMCLANCNFSM5ZLW2FBA>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
<https://netanel.ml>

**Reply by crowdstrikedcs** [reply]
Hi Nate, at this time the ability to create scheduled reports can only be done in the Falcon UI. There is no API for creating scheduled reports.
